packages/components/src/internal/components/domainproperties/TextChoiceOptions.tsx

```diff
@@ -182,35 +250,52 @@ export const TextChoiceOptionsImpl: FC<ImplProps> = memo(props => {
+                                    <DisableableInput
                                         className="form-control full-width"
-                                        disabled={currentInUse}
+                                        disabledMsg={currentLocked ? LOCKED_TIP : undefined}
                                         name="value"
                                         onChange={onValueChange}
                                         placeholder="Enter a text choice value"
-                                        type="text"
+                                        title={LOCKED_TITLE}
                                         value={currentValue ?? ''}
                                     />
                                 </div>
-                                <div>
+                                <div className="domain-field-padding-bottom">
                                     <DisableableButton
                                         bsStyle="default"
-                                        disabledMsg={currentInUse ? IN_USE_TIP : undefined}
+                                        disabledMsg={currentLocked ? LOCKED_TIP : currentInUse ? IN_USE_TIP : undefined}
                                         onClick={onDelete}
-                                        title={IN_USE_TITLE}
+                                        title={currentLocked ? LOCKED_TITLE : IN_USE_TITLE}
                                     >
                                         <span className="fa fa-trash" />
                                         <span>&nbsp;Delete</span>
                                     </DisableableButton>
                                     <Button
                                         bsStyle="success"
                                         className="pull-right"
-                                        disabled={currentValue === validValues[selectedIndex]}
+                                        disabled={
+                                            currentError !== undefined ||
+                                            currentValue === selectedValue ||
+                                            currentValue === ''
+                                        }
                                         onClick={onApply}
                                     >
                                         Apply
                                     </Button>
                                 </div>
+                                {fieldValueUpdates[selectedValue] !== undefined &&
+                                    selectedValue !== fieldValueUpdates[selectedValue] && (
+                                        <Alert bsStyle="info">
+                                            {Utils.pluralize(
+                                                fieldValues[fieldValueUpdates[selectedValue]].count,
+                                                'row',
+                                                'rows'
```

Comment:
I guess we don't have the domain noun here to use instead of "row"?

Reply:
We don't, unfortunately. There is a "helpNoun" property but that doesn't quite fit here.